src/components/virtualSessions/VirtualSession.tsx

```diff
@@ -0,0 +1,202 @@
+`
+
+const ConfirmButton = styled.button`
+  background-color: #FCC600;
+  border: none;
+  border-radius: 4px;
+  color: #000000;
+  padding: 15px 62px;
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+  font-size: 16px;
+  margin: 4px 2px;
+
+  :disabled {
+    border: 1px solid #999999;
+    background-color: #cccccc;
+    color: #666666;
+  }
+`
+
+interface ActionButtonProps {
+  active: boolean;
+}
+
+const div: StyledFunction<
+  ActionButtonProps & React.HTMLProps<HTMLInputElement>
+> = styled.div;
+
+const ActionButton = div`
+  font-family: 'Lato', sans-serif;
+  border: none;
+  border-radius: 4px;
+  padding: 15px 32px;
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+  font-size: 16px;
+  cursor: pointer;
+  background-color: ${(props: ActionButtonProps) =>
+    props.active ? "#fcc600" : "#cccccc"};
+  color: ${(props: ActionButtonProps) =>
+    props.active ? "#000000" : "#666666"};
+`;
+
+interface IVirtualSessionParams {
+  id: string;
+}
+
+interface IVirtualSessionState {
+  virtualSession?: VirtualSessionInterface;
+  timePreferences?: string[];
+  preferenceSelected: number;
+  virtualSessionValid: boolean;
+  actionButtonText: string;
+}
+
+export class VirtualSession extends React.Component<RouteComponentProps<IVirtualSessionParams>, IVirtualSessionState> {
+  public apiService = new ServiceResolver().ApiService();
+  constructor(props: RouteComponentProps<IVirtualSessionParams>) {
+    super(props);
+    this.state = {
+      virtualSession: undefined,
+      timePreferences: undefined,
+      preferenceSelected: 0,
+      virtualSessionValid: true,
+      actionButtonText: 'Confirm'
+    }
+  }
+
+  public async componentDidMount() {
+    const virtualSession = await this.apiService.getVirtualSession(this.props.match.params.id);
+    const virtualSessionValid = virtualSession.volunteerId === "00000000-0000-0000-0000-000000000000";
+    const timePreferences = [virtualSession.timePreferenceOne!, virtualSession.timePreferenceTwo!, virtualSession.timePreferenceThree!];
+    this.setState({
+      virtualSession,
+      timePreferences,
+      virtualSessionValid
+    });
+  }
+
+  public handleTimePreferenceSelect = (preferenceSelected: number) => {
+    this.setState({
+      preferenceSelected
+    });
+  }
+
+  public handleConfirmClick = async () => {
+    const memberNotPresent = SessionStorageHelper.GetJwt()! === undefined;
+    if (memberNotPresent) {
+      return;
+    }
+
+    const volunteerId = JwtTokenHelper.decodeMemberId(SessionStorageHelper.GetJwt()!.token);
+
+    const { virtualSession, timePreferences, preferenceSelected } = this.state;
+    virtualSession!.volunteerId = volunteerId;
+    virtualSession!.timePreferenceSelected = timePreferences![preferenceSelected];
+
+    try {
+      await this.apiService.updateVirtualSession(this.props.match.params.id, virtualSession!);
+      this.setState({
+        virtualSessionValid: false,
```

Comment:
I think because I'm using the test user, it tells me all sessions have been confirmed by other volunteers.

Reply:
yeah you're right. the mock response being returned (`getVirtualSession.ts`) is returned with the default id `93b2d860-df81-4688-aaa4-23b4e9da0f00`. If instead we return the default value for an empty guid, `00000000-0000-0000-0000-000000000000`, then the virtual session is yet to be assigned to a volunteer. Only in that case will you see the button enabled.